crates/weaver_semconv/src/any_value.rs

```diff
@@ -0,0 +1,690 @@
+    },
+
+    /// An enum definition type.
+    Enum {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+
+        /// Set to false to not accept values other than the specified members.
+        /// It defaults to true.
+        #[serde(default = "default_as_true")]
+        allow_custom_values: bool,
+        /// List of enum entries.
+        members: Vec<EnumEntriesSpec>,
+    }    
+}
+
+/// The Common Value specification for properties associated with an "AnyValue", this
+/// is similar to the current `AttributeSpec` as at the proto level an Attribute
+/// is defined as a "KeyValue".
+/// While this is (currently) a duplication of the existing AttributeSpec, this is
+/// to reduce the size of the change set.
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash, JsonSchema)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "snake_case")]
+pub struct AnyValueCommonSpec {
+    /// String that uniquely identifies the enum entry.
+    pub id: String,
+    /// A brief description of the value
+    #[serde(skip_serializing_if = "String::is_empty")]
+    #[serde(default)]
+    pub brief: String,
+    /// A more elaborate description of the value.
+    /// It defaults to an empty string.
+    #[serde(skip_serializing_if = "String::is_empty")]
+    #[serde(default)]
+    pub note: String,
+    /// Specifies the stability of the value.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub stability: Option<Stability>,
+    /// Sequence of examples for the value or single example
+    /// value. If only a single example is provided, it can
+    /// directly be reported without encapsulating it
+    /// into a sequence/dictionary.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub examples: Option<Examples>,
+    /// Specifies if the field is mandatory. Can be "required",
+    /// "conditionally_required", "recommended" or "opt_in". When omitted,
+    /// the field is "recommended". When set to
+    /// "conditionally_required", the string provided as <condition> MUST
+    /// specify the conditions under which the field is required.
+    pub requirement_level: RequirementLevel,
+    /// Specifies if the body field is deprecated. The string
+    /// provided as <description> MUST specify why it's deprecated and/or what
+    /// to use instead. See also stability.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub deprecated: Option<String>,
+}
+
+/// Implements a human readable display for AnyValueType.
+impl Display for AnyValueSpec {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            AnyValueSpec::Map { fields, .. } => {
+                let entries = fields
+                    .iter()
+                    .map(|v| format!("{}", v))
+                    .collect::<Vec<String>>()
+                    .join(", ");
+                write!(f, "map<{}>{{ {} }}", self.id(), entries)
```

Comment:
Discussion point: 
Slight difference from the previous as I'm using the `id` to decorate the "name" of the `map` (and `enum`)

Reply:
Ok, as part of "testing" this in the Events PR this "Display" appears to be used for generated registry `json` so this "can't" use this decoration.

Reply:
to further the discussion point, I've currently kept this Display and added a new `type_name()` to "populate" the resolved JSON with the "short" version of the names.

Will update this PR shortly with the changes.